addons/project_account/models/project_project.py

```diff
@@ -0,0 +1,144 @@
+                analytic_contribution = moves_read['analytic_distribution'][str(self.analytic_account_id.id)] / 100.
+                move_ids.add(moves_read['move_id'])
+                if moves_read['parent_state'] == 'draft':
+                    if moves_read['move_type'] == 'in_invoice':
+                        amount_to_invoice -= price_subtotal * analytic_contribution
+                    else:  # moves_read['move_type'] == 'in_refund'
+                        amount_to_invoice += price_subtotal * analytic_contribution
+                else:  # moves_read['parent_state'] == 'posted'
+                    if moves_read['move_type'] == 'in_invoice':
+                        amount_invoiced -= price_subtotal * analytic_contribution
+                    else:  # moves_read['move_type'] == 'in_refund'
+                        amount_invoiced += price_subtotal * analytic_contribution
+            # don't display the section if the final values are both 0 (bill -> vendor credit)
+            if amount_invoiced != 0 or amount_to_invoice != 0:
+                costs = profitability_items['costs']
+                section_id = 'other_purchase_costs'
+                bills_costs = {
+                    'id': section_id,
+                    'sequence': self._get_profitability_sequence_per_invoice_type()[section_id],
+                    'billed': amount_invoiced,
+                    'to_bill': amount_to_invoice,
+                }
+                if with_action:
+                    bills_costs['action'] = self._get_action_for_profitability_section(list(move_ids), section_id)
+                costs['data'].append(bills_costs)
+                costs['total']['billed'] += amount_invoiced
+                costs['total']['to_bill'] += amount_to_invoice
+
+    def _get_action_for_profitability_section(self, record_ids, name):
+        self.ensure_one()
+        args = [name, [('id', 'in', record_ids)]]
+        if len(record_ids) == 1:
+            args.append(record_ids[0])
+        return {'name': 'action_profitability_items', 'type': 'object', 'args': json.dumps(args)}
+
+    def action_profitability_items(self, section_name, domain=None, res_id=False):
+        if section_name in ['other_revenues', 'other_costs']:
+            action = self.env["ir.actions.actions"]._for_xml_id("analytic.account_analytic_line_action_entries")
+            action['domain'] = domain
+            action['context'] = {
+                'group_by_date': True,
+            }
+            if res_id:
+                action['views'] = [(False, 'form')]
+                action['view_mode'] = 'form'
+                action['res_id'] = res_id
+            else:
+                pivot_view_id = self.env.ref('project_account.project_view_account_analytic_line_pivot').id
+                graph_view_id = self.env.ref('project_account.project_view_account_analytic_line_graph').id
```

Comment:
Since we just need id, we could use `self.env['ir.model.data']._xmlid_to_res_id` instead of `self.env.ref`. 🙂 
It is equivalent except `ref` returns a recordset. 🙂